blog/_posts/2022-08-08-managing-tunnels-with-argocd.md

```diff
@@ -0,0 +1,303 @@
+spec:
+  generators:
+  - list:
+      elements:
+      - name: postgresql-customer1-tunnel
+        valuesFile: values-postgresql-customer1.yaml
+  template:
+    metadata:
+      name: '{{name}}'
+    spec:
+      project: default
+      source:
+        repoURL: https://github.com/welteki/inlets-server-argocd-demo.git
+        targetRevision: HEAD
+        path: charts/inlets
+        helm:
+          valueFiles:
+            - '{{valuesFile}}'
+      destination:
+        server: https://kubernetes.default.svc
+        namespace: default
+      syncPolicy:
+        automated:
+          selfHeal: true
+```
+
+This ApplicationSet uses the [List generator](https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Generators-List/) which generates parameters based on a fixed list of arbitrary key/value pairs. In this case it contains values for the name of our inlets servers and the values file that Helm needs to use for the deployment.
+
+In the application template section the fields like `{% raw %}{{name}}{% endraw %}` and `{% raw %}{{valuesFile}}{% endraw %}` are populated via parameters that will be replaced by the generator.
+
+The process of deploying an additional inlets tunnel server with the ApplicationSet now looks like this:
+- Add an additional token secret and `values.yaml` file to the chart.
+- Add the server to the list of key/value pairs in the ApplicationSet generators section.
+
+    ```yaml
+    generators:
+    - list:
+        elements:
+        - name: postgresql-customer1-tunnel
+          valuesFile: values-postgresql-customer1.yaml
+          # Add second server
+        - name: mysql-customer2-tunnel
+          valuesFile: values-mysql-customer2.yaml
+    ```
+
+- Commit and push your changes
+- Sync changes manually in the UI or wait for Argo CD to automatically detect the drift in the repository.
+- A new Argo CD application for a tunnel server should be created.
+
+## Wrapping up
+We showed you how Argo CD can be used to simplify deploying and managing multiple Inlets tunnel servers. We also demonstrated one of the many possible ways to manage secret authentication with Argo CD using [Bitnami Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets).
+
+Some resources like the nginx ingress, Sealed Secrets controller and the Argo CD application for tunnel servers where installed manually with `arkade` or by running `kubectl apply`. The [app of apps pattern](https://argo-cd.readthedocs.io/en/stable/operator-manual/cluster-bootstrapping/) can be used to automate all these steps and bootstrap your entire cluster with Argo CD.
+
+In the tutorial the ApplicationSet is used to deploy all Argo CD applications to a single cluster. Additional parameters can be added to the template to deploy Inlets tunnel servers to multiple Kubernetes clusters. The examples from the [Argo CD user guide](https://argo-cd.readthedocs.io/en/stable/user-guide/application-set/) can be used as a reference for this.
```

Comment:
Perhaps a short section:

You may also like

Linking to the examples of ArgoCD on the OpenFaaS blog.